tools/DevDiagnostics/DevHome.DevDiagnostics/ViewModels/WERPageViewModel.cs

```diff
@@ -199,55 +213,66 @@ public void ResetBucketingTool()
         }
     }
 
-    [RelayCommand]
-    private void RunAsAdmin()
+    public void OpenCab(string file)
     {
-        if (TargetAppData.Instance.TargetProcess is not null)
+        if (File.Exists(file))
         {
-            CommonHelper.RunAsAdmin(TargetAppData.Instance.TargetProcess.Id, nameof(WERPageViewModel));
+            ProcessStartInfo startInfo = new ProcessStartInfo
+            {
+                FileName = file,
+                UseShellExecute = true,
+            };
+
+            Process.Start(startInfo);
         }
     }
 
-    public void ChangeLocalCollectionForApp(bool enable)
+    [RelayCommand]
+    public void ToggleLocalCabCollection()
     {
-        Process? process = TargetAppData.Instance.TargetProcess;
-
-        if (process is null || process.ProcessName is null)
-        {
-            return;
-        }
-
-        string app = process.ProcessName + ".exe";
+        string? attachedApp = TargetAppData.Instance?.TargetProcess?.ProcessName ?? null;
 
-        if (enable == _werHelper.IsCollectionEnabledForApp(app))
+        if (attachedApp is null)
         {
-            // No change, could be initialization of the UI
             return;
         }
 
-        Debug.Assert(RuntimeHelper.IsCurrentProcessRunningAsAdmin(), "Changing the local dump collection for an app can only happen when running as admin.");
-
-        if (enable)
+        ThreadPool.QueueUserWorkItem((o) =>
         {
-            _werHelper.EnableCollectionForApp(app);
-        }
-        else
-        {
-            _werHelper.DisableCollectionForApp(app);
-        }
-    }
+            App.Log("ToggleLocalCabCollection", LogLevel.Measure);
 
-    public void OpenCab(string file)
-    {
-        if (File.Exists(file))
-        {
-            ProcessStartInfo startInfo = new ProcessStartInfo
+            try
             {
-                FileName = file,
-                UseShellExecute = true,
-            };
-
-            Process.Start(startInfo);
-        }
+                FileInfo fileInfo = new FileInfo(Environment.ProcessPath ?? string.Empty);
+
+                var startInfo = new ProcessStartInfo()
+                {
+                    FileName = "EnableLocalCabCollection.exe",
+                    Arguments = attachedApp + ".exe",
+                    UseShellExecute = true,
+                    WorkingDirectory = fileInfo.DirectoryName,
+                    Verb = "runas",
+                };
+
+                var process = Process.Start(startInfo);
+                if (process is not null)
+                {
+                    // Wait for the process to update registry keys
+                    process.WaitForExit();
```

Comment:
there is no way to tell if the separate exe was successful.  At a minimum I would hope we could use the process exit code.

Reply:
Just pushed an update that compares the "enabled" status before and after the launch, and it they are the same (meaning we were unsuccessful in toggling the setting) we'll log a telemetry point.

I'm hoping if this is unsuccessful, we'll see it in a Watson crash from the helper exe.